tools/check-directives.sh

```diff
@@ -1,37 +1,71 @@
 #!/usr/bin/env bash
 # SPDX-License-Identifier: LGPL-2.1-or-later
-set -e
+set -eu
 
-which perl &>/dev/null || exit 77
+SOURCE_ROOT="${1:?Missing argument: project source root}"
+BUILD_ROOT="${2:?Missing argument: project build root}"
+
+command -v gawk &>/dev/null || exit 77
 
 function generate_directives() {
-    perl -aF'/[\s,]+/' -ne '
-        if (my ($s, $d) = ($F[0] =~ /^([^\s\.]+)\.([^\s\.]+)$/)) { $d{$s}{"$d="} = 1; }
-        END { while (my ($key, $value) = each %d) {
-            printf "[%s]\n%s\n", $key, join("\n", keys(%$value))
-        }}' "$1"
+    gawk -v sec_rx="${2:-""}" -v unit_type="${3:-""}" '
```

Comment:
I wonder what was wrong with perl? :-)

Reply:
Nothing in particular, actually, but it's been a while since I wrote anything in perl, so it was much faster to write it in awk than to re-learn perl. :-)